mne/tests/test_filter.py

```diff
@@ -192,6 +192,11 @@ def test_filters():
     assert_true(iir_params['a'].size - 1 == 4)
     assert_true(iir_params['b'].size - 1 == 4)
 
+    # check for n-dimensional case
+    a = np.random.randn(2, 2, 2, 2)
+    assert_raises(ValueError, band_pass_filter, a, sfreq, Fp1=4, Fp2=8,
+                  picks=np.array([0, 1]))
```

Comment:
can you add a test that filtering a 3d array of shape (1, p, q) yields the same result as the (p, q) array when setting picks=[0]?